### DISCUSSION TITLE
How to sort scroll?

### DISCUSSION THREAD

**Discussion by msj121** [qdrant/qdrant#2905] [Q&A]
It should be possible to sort results by a payload qualifier, even if this is limited to scroll.... If I scroll all restaurants that serve pizza, why can't I sort by when it was established, or the price etc.... Is this part of the future roadmap, because I readily need this. I assume others do too... Is there a current work around?

**Comment by skiiiks** [comment]
Hi @msj121 .
Only workaround is to make a batch search query and apply multiple conditions.

It's a pity nobody of Qdrant replies.

I believe that 100% technological products such as these, where all "customers" are developers, should have an exquisite technical support.

It is precisely us developers who recommend this type of solutions to other colleagues, and above all for the activity and agility of the community that there is supporting and helping.

But if the company itself does not have a resource so that there are no questions like yours unanswered for 3 weeks, is that they do not know who their customers really are.

**Reply by timvisee** [reply]
> It's a pity nobody of Qdrant replies.
> 
> I believe that 100% technological products such as these, where all "customers" are developers, should have an exquisite technical support.
> 
> It is precisely us developers who recommend this type of solutions to other colleagues, and above all for the activity and agility of the community that there is supporting and helping.
> 
> But if the company itself does not have a resource so that there are no questions like yours unanswered for 3 weeks, is that they do not know who their customers really are.

We are all very busy building the best product possible. That takes a lot of time. While we're not always able to answer all questions on GitHub, we do have a very active Discord community over at: https://qdrant.to/discord

**Comment by timvisee** [comment]
> If I scroll all restaurants that serve pizza, why can't I sort by when it was established, or the price etc....

This is a limitation because of how scroll is implemented. Scroll goes over all point IDs in order, allowing you to fetch all points from a collection.

It currently does not support a different sorting approach as it would require Qdrant to sort all point IDs beforehand. While this would work fine if you'd have 100 points, this doesn't scale well to billions of points.

In your case, would it be possible to fetch all results and sort in your client instead?

Also, could you please elaborate further on why you'd need this? It sounds like a type of query where a classic relational database would work better.

**Comment by skiiiks** [comment]
> > It's a pity nobody of Qdrant replies.
> > I believe that 100% technological products such as these, where all "customers" are developers, should have an exquisite technical support.
> > It is precisely us developers who recommend this type of solutions to other colleagues, and above all for the activity and agility of the community that there is supporting and helping.
> > But if the company itself does not have a resource so that there are no questions like yours unanswered for 3 weeks, is that they do not know who their customers really are.
> 
> We are all very busy building the best product possible. That takes a lot of time. While we're not always able to answer all questions on GitHub, we do have a very active Discord community over at: https://qdrant.to/discord

I have also emails and messages on discord not replied :-D

**Reply by timvisee** [reply]
> I have also emails and messages on discord not replied :-D

Please feel free to link them here, and I'd be happy to answer.

If you'd like to get priority support you can sign a SLA contract with us.

**Reply by skiiiks** [reply]
I'm not pretending any priority, just to tell you that you should listen more to your customers.
It's more important to listen than being busy building something maybe your customers don't want.
Customers feedback have higher priority than building something.
But it's your company ;-)

I just told you that, when you have full technological products (like this one) is important to be CLOSE and to answer FAST to your customers.

@msj121 waited for 3 weeks (and he was lucky I answered him)

**Reply by timvisee** [reply]
We have a lot of fantastic customers, and they give us a lot of valuable feedback. Large and small. We don't build things that have no use case. There's a lot, so we have to prioritize. Don't forget that you get all this for free.

If something important is missing for your use case, you may always consider to contribute this yourself. We're open source.

Let's end this discussion here as it's not constructive for the original question.
